src/main/java/it/gov/pagopa/payment/service/payment/common/CommonAuthCodeExpiration.java

```diff
@@ -0,0 +1,55 @@
+package it.gov.pagopa.payment.service.payment.common;
+
+import it.gov.pagopa.common.web.exception.ClientException;
+import it.gov.pagopa.payment.connector.rest.reward.RewardCalculatorConnector;
+import it.gov.pagopa.payment.enums.SyncTrxStatus;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.repository.TransactionInProgressRepository;
+import it.gov.pagopa.payment.utils.AuditUtilities;
+import org.springframework.http.HttpStatus;
+
+public class CommonAuthCodeExpiration extends BaseCommonCodeExpiration{
+    private final long authorizationExpirationMinutes;
+    private final TransactionInProgressRepository transactionInProgressRepository;
+    private final RewardCalculatorConnector rewardCalculatorConnector;
+    protected CommonAuthCodeExpiration(AuditUtilities auditUtilities,
+                                       String channel,
+                                       long authorizationExpirationMinutes,
+                                       TransactionInProgressRepository transactionInProgressRepository,
+                                       RewardCalculatorConnector rewardCalculatorConnector) {
+        super(auditUtilities, channel);
+        this.authorizationExpirationMinutes = authorizationExpirationMinutes;
+        this.transactionInProgressRepository = transactionInProgressRepository;
+        this.rewardCalculatorConnector = rewardCalculatorConnector;
+    }
+
+    @Override
+    protected long getExpirationMinutes() {
+        return authorizationExpirationMinutes;
+    }
+
+    @Override
+    protected TransactionInProgress findExpiredTransaction(String initiativeId, long expirationMinutes) {
+        return transactionInProgressRepository.findAuthorizationExpiredTransaction(initiativeId, expirationMinutes);
+    }
+
+    @Override
+    protected TransactionInProgress handleExpiredTransaction(TransactionInProgress trx) {
+        if (trx.getStatus().equals(SyncTrxStatus.IDENTIFIED)) {
```

Comment:
put a TODO to remember to BRCODE flow to change this in order to check also for CREATED status